### PR TITLE
Char, Hex, Base64, ASCII, and Unicode

### PR DESCRIPTION
**Pull request by RuiGuedes**
Covers the following points of #45: `Char`, `Hex`, `Base64`, `ASCII`, and `Unicode`. Besides the implementation of such features, it was also made the refactor of the `ArbitrartyString`.

### CODE REVIEW

**Comment by RuiGuedes** [comment]
This PR is almost finished. Currently, the only thing missing is changing the `string()` function to be based on `CharArbitraries` instead of its current state. However, such modification depends on the `MappedArbitrary` `canGenerate()` function, which limitation is addressed on PR #153.

Even though such modification could be made in another PR I see more fit to be done on this one to encapsulate all the logic associated with `strings` and `chars` on this PR. Therefore, this PR is on hold until PR #153 is merged.

**Comment by RuiGuedes** [comment]
Despite this PR being marked as **draft** I believe that it is almost finished. The only thing missing is a set of tests to test the newly implemented functions. As such @ruippeixotog @arestivo @hugoferreira feel free to start giving me your feedback.

**Comment by RuiGuedes** [comment]
Depends on #153

File: src/arbitraries/index.ts
Comment (RuiGuedes) [reply src/arbitraries/index.ts:50]: I agree with and in fact, @hugoferreira has already stated the same. My doubt is if want to make such alterations in another PR or include them in this one. Let me know what you think guys.

Comment (ruippeixotog) [reply src/arbitraries/index.ts:50]: Given that this PR is introducing these functions, I see no reason why we shouldn't put them in the right place from the start :+1:

Comment (RuiGuedes) [reply src/arbitraries/index.ts]: @ruippeixotog @hugoferreira Already added a new `string.ts` file for encapsulating the logic associated with characters and strings. Do you guys have more suggestions to be made?

File: src/arbitraries/index.ts
Comment (hugoferreira) [inline src/arbitraries/index.ts:16]: nit: extra `,`

**Review by hugoferreira — CHANGES_REQUESTED** [review]

File: src/arbitraries/index.ts
Comment (RuiGuedes) [reply src/arbitraries/index.ts]: Done!

**Pull request merged**
